src/Components/LoginForm.tsx

```diff
@@ -0,0 +1,233 @@
+import React, {
+  FormEvent,
+  useContext,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { AppContext } from '../AppContext';
+import { findUser, addUsers } from '../api/todos';
+import { Notification } from './Notification';
+
+const addPrevUser = (prevEmail: string) => {
+  const getPrevUsers: string[] | 0 = JSON
+    .parse(localStorage.getItem('prevusers') || '0');
+
+  if (getPrevUsers !== 0) {
+    if (getPrevUsers.length === 5) {
+      getPrevUsers.pop();
+    }
+
+    if (getPrevUsers.filter(mail => mail === prevEmail).length === 0) {
+      getPrevUsers.unshift(prevEmail);
+    }
+
+    localStorage.setItem('prevusers', JSON.stringify(getPrevUsers));
+  } else {
+    localStorage.setItem('prevusers', JSON.stringify([prevEmail]));
+  }
+};
+
+export const LoginForm: React.FC = () => {
+  const { setUser } = useContext(AppContext);
+  const [email, setEmail] = useState('');
+  const [name, setName] = useState('');
+  const [newUser, setNewUser] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [emailDisabled, setEmailDisabled] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [showPrevUsers, setShowPrevUsers] = useState(false);
+  const hasErrorFromServer = !!errorMessage;
+
+  const ref = useRef<HTMLInputElement>(null);
+
+  let tempEmail = '';
+
+  const prevUsers: string[] | 0 = JSON
+    .parse(localStorage.getItem('prevusers') || '0');
+
+  const toshowPrevUsers = useMemo(() => {
+    return prevUsers !== 0 ? prevUsers
+      .filter(preMail => {
+        return email !== '' ? preMail.includes(email) : true;
+      }) : 0;
+  }, [email]);
```

Comment:
Check if it'll work as well
```suggestion
  const toshowPrevUsers = useMemo(() => 
    prevUsers !== 0
      ? prevUsers.filter(preMail => !email.length || preMail.includes(email))
      : 0, 
    [email]
  );
```